### PR TITLE
[eslint-patch] fix: add empty export for module type detection

### LINKED ISSUE #3343: [eslint-patch] warning from vite bundling as cannot determine if CJS or ESM
## Summary

When the Vite tool is being used in a package, it performs automatic dependency discovery during builds. During this process it attempts to determine whether each package in the project is CommonJS or ESM. As this is automatic for all the project dependencies, it discovers `eslint-patch` (even though it will not included in the final build).

Vite outputs the following warning during bundling:

```
@rushstack/eslint-patch doesn't appear to be written in CJS, but also doesn't appear to be a
valid ES module (i.e. it doesn't have "type": "module" or an .mjs extension for the entry point).
Please contact the package author to fix.
```

## Problem

The entry point for `eslint-patch` throws an error to explain the package does not have a default entry point. However there is nothing in this code that identifies it as CommonJS, causing Vite to output the warning.

## Fix

Adding a blank export provides sufficient information for Vite to determine this is CommonJS, and the warning is no longer output. The intended error will still be thrown if the package is directly imported.

## PR

PR will be submitted.

## Standard questions

Please answer these questions to help us investigate your issue more quickly:

| Question | Answer |
| -------- | -------- |
| `@rushstack/eslint-patch` version? | 1.1.2 |
| Would you consider contributing a PR? | Yes |

### PR DESCRIPTION
**Pull request by tom-fletcher**
## Summary

Adds a blank `export = {}` to the `eslint-patch` entry point to allow other tools (specifically Vite) to automatically determine this is Common JS.

Fixes #3343

## Details

Adding a blank export provides enough information for Vite to determine this is indeed CommonJS.

This entry point is only used to throw an error to warn the user not to directly import the package. No breaking change. No performance impacts.

## How it was tested

Re-built my project with Vite, the warning mentioned in #3343 was no longer output.

Imported the package directly, and validated the usage error was still thrown as expected.

### CODE REVIEW

File: eslint/eslint-patch/src/usage.ts [hunk 1]
```diff
@@ -5,3 +5,5 @@ throw new Error(
   'The @rushstack/eslint-patch package does not have a default entry point.' +
     ' See README.md for usage instructions.'
 );
+
+export = {};
```
Comment (dmichon-msft) [inline eslint/eslint-patch/src/usage.ts:9]:
```suggestion
export {};
```
TypeScript will emit code to identify it is as the module type declared in tsconfig.json, which in this case is CommonJS.

Comment (octogonz) [reply eslint/eslint-patch/src/usage.ts:9]: +1  

`export =` is deprecated.

Comment (tom-fletcher) [reply eslint/eslint-patch/src/usage.ts]: Thanks! I've checked this continues to work as expected with the vite build, and can confirm the warning doesn't appear.

**Review by octogonz — APPROVED** [review]

**Comment by octogonz** [comment]
Thanks for making this fix!

**Pull request merged**